src/oc_erchef/apps/oc_chef_wm/src/oc_chef_action_insights.erl

```diff
@@ -0,0 +1,27 @@
+%% Copyright 2012-2014 Chef Software, Inc. All Rights Reserved.
+%%
+%% This file is provided to you under the Apache License,
+%% Version 2.0 (the "License"); you may not use this file
+%% except in compliance with the License.  You may obtain
+%% a copy of the License at
+%%
+%%   http://www.apache.org/licenses/LICENSE-2.0
+%%
+%% Unless required by applicable law or agreed to in writing,
+%% software distributed under the License is distributed on an
+%% "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+%% KIND, either express or implied.  See the License for the
+%% specific language governing permissions and limitations
+%% under the License.
+%%
+
+-module(oc_chef_action_insights).
+
+-export([ingest/2]).
+
+-spec ingest(ShouldSend :: boolean(), Data :: binary()) -> ok.
+ingest(true, Data) ->
+    InsightsLogDir = envy:get(oc_chef_wm, insights_log, string),
+    InsightsLog = filename:join(InsightsLogDir, "insights"),
+    ok = file:write_file(InsightsLog, Data, [append]);
```

Comment:
My only concern with this is that `file:write_file` is most likely going to serialize access to this file through a single gen_server or process mailbox. We've seen issues with this in bookshelf doing synchronous request logging that caused the Erlang VM to hang, balloon in memory, and eventually crash.